public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/url_builder.ts

```diff
@@ -7,20 +7,16 @@ export default class UrlBuilder {
     resourceName: string,
     apiVersion: string
   ) {
-    if ((metricDefinition.match(/\//g) || []).length > 1) {
-      const rn = resourceName.split('/');
-      const service = metricDefinition.substring(metricDefinition.lastIndexOf('/') + 1);
-      const md = metricDefinition.substring(0, metricDefinition.lastIndexOf('/'));
-      return (
-        `${baseUrl}/${subscriptionId}/resourceGroups/${resourceGroup}/providers/${md}/${rn[0]}/${service}/${rn[1]}` +
-        `/providers/microsoft.insights/metricNamespaces?api-version=${apiVersion}`
-      );
+    var metricDefinitionArray = metricDefinition.split('/');
+    const resourceNameArray = resourceName.split('/');
+    const provider = metricDefinitionArray.shift();
+    var urlArray = [baseUrl, subscriptionId, 'resourceGroups', resourceGroup, 'providers', provider];
+    for (var i in metricDefinitionArray) {
```

Comment:
these three `var` should be `const`

Reply:
https://www.typescriptlang.org/docs/handbook/variable-declarations.html#const-declarations
>This should not be confused with the idea that the values they refer to are immutable.

Aha, that's exactly the misunderstanding I had, thanks!

Reply:
I have now fixed the datasource url builder as well :)